framework/src/utils/SlepcSupport.C

```diff
@@ -0,0 +1,52 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef SLEPCSUPPORT_H
+#define SLEPCSUPPORT_H
+
+#include "libmesh/libmesh.h"
+
+#ifdef LIBMESH_HAVE_SLEPC
+
+#include "SlepcSupport.h"
+// MOOSE includes
+#include "MultiMooseEnum.h"
+#include "InputParameters.h"
+
+namespace Moose
+{
+namespace SlepcSupport
+{
+
+  InputParameters
+  getPetscValidParams()
+  {
+    InputParameters params = emptyInputParameters();
+
+    MooseEnum eigenvalue_problem_type("HEP NHEP GHEP GNHEP GHIEP");
```

Comment:
> does missing the second argument means that there is no default value for this moose enum and the later parameter is required?

We do not set the default parameter here. If users do not  setup anything in the input file, we have a default parameter setup from them in the solver.  The reason I do not like a default setup here is that I want to use` params.isParamValid("")`.